ruby/vampires.rb

```diff
@@ -1,54 +1,42 @@
-vamp_likelihood = 5
-
-puts "What is your name?"
-name = gets.chomp
-  if name == "Drake Cula" || "Tu Fang"
-    vamp_likelihood += 10
-end
-  
-puts "What year were you born?"
-year = gets.chomp
-age = 2017 - year.to_i
-  if age <= 60
-    vamp_likelihood -= 2
-  else 
-    vamp_likelihood += 2
-end
-
-puts "Do you want garlic bread (y/n)?"
-bread = gets.chomp
-  if bread = "y"
-    vamp_likelihood -= 1
-  else vamp_likelihood += 1
-end
-
-puts "Would you like to enroll in the Company's health insurance? (y/n)"
-health = gets.chomp
-  if health = "y"
-      vamp_likelihood -= 1
-    else vamp_likelihood += 1
-end
-
-result = case vamp_likelihood
-   when 2..3 then "Probably not a vampire"
-   when 4..5 then "Probably a vampire"
-   when 6..10 then "Almost certainly a vampire"
-   when 10..15 then "Definitely a vampire"
-   else "Results inconclusive"
+puts "How many employees need to be processed?"
+employees = gets.chomp
+employees = employees.to_i
+$count = 1
+$valid_input = "invalid"
+
+while $count <= employees
+  puts "Processing employee #{$count}/#{employees}"
+	puts "What is your name?"
+	name = gets.chomp
+	puts "What year were you born?"
+	year = gets.chomp
+	age = 2017 - year.to_i
+	puts "Do you want garlic bread (y/n)?"
+	bread = gets.chomp
+	puts "Would you like to enroll in the Company's health insurance? (y/n)"
+	health = gets.chomp
+  puts "Please name your allergies one at a time. Type 'done' when finished"
+	$allergies = gets.chomp
+    until $allergies == "sunshine" || $allergies == "done"
+    	puts "Do you have any other allergies?"
+    	puts "Please name your allergies one at a time. Type 'done' when finished"
+    	$allergies = gets.chomp
+    end
+	if name == "Drake Cula" || name == "Tu Fang" || $allergies == "sunshine"
```

Comment:
Instead of tying all of these checks together, we could write them all as separate `if` checks. Then we can store the return string in a variable and print that to the console at the end of the loop. These checks do not depend on each other, so no need to link them.